databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,346 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import csv
+import ctypes
+import logging
+import time
+from io import open
+from typing import (
+    Dict, List, Set,
+)
+
+import neo4j
+import pandas
+from jinja2 import Template
+from neo4j import GraphDatabase, Neo4jDriver
+from neo4j.api import (
+    SECURITY_TYPE_SECURE, SECURITY_TYPE_SELF_SIGNED_CERTIFICATE, parse_neo4j_uri,
+)
+from pyhocon import ConfigFactory, ConfigTree
+
+from databuilder.models.graph_serializable import (
+    NODE_KEY, NODE_LABEL, RELATION_END_KEY, RELATION_END_LABEL, RELATION_REVERSE_TYPE, RELATION_START_KEY,
+    RELATION_START_LABEL, RELATION_TYPE,
+)
+from databuilder.publisher.base_publisher import Publisher
+from databuilder.publisher.publisher_config_constants import (
+    Neo4jCsvPublisherConfigs, PublishBehaviorConfigs, PublisherConfigs,
+)
+from databuilder.utils.publisher_utils import (
+    chunkify_list, create_neo4j_node_key_constraint, create_props_param, execute_statement, get_props_body_keys,
+    list_files,
+)
+
+# Setting field_size_limit to solve the error below
+# _csv.Error: field larger than field limit (131072)
+# https://stackoverflow.com/a/54517228/5972935
+csv.field_size_limit(int(ctypes.c_ulong(-1).value // 2))
+
+# Required columns for Node
+NODE_REQUIRED_KEYS = {NODE_LABEL, NODE_KEY}
+# Required columns for Relationship
+RELATION_REQUIRED_KEYS = {RELATION_START_LABEL, RELATION_START_KEY,
+                          RELATION_END_LABEL, RELATION_END_KEY,
+                          RELATION_TYPE, RELATION_REVERSE_TYPE}
+
+DEFAULT_CONFIG = ConfigFactory.from_dict({Neo4jCsvPublisherConfigs.NEO4J_TRANSACTION_SIZE: 500,
```

Comment:
make this a nice round 1k i think. Don't have strong reasoning, but 500 seems small compared to the online blogs that talk about UNWIND.